crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2976,6 +2979,60 @@ impl Room {
             .await?;
         Ok(())
     }
+
+    /// Subscribe to live location sharing events for this room.
+    ///
+    /// The returned receiver will receive a new event for each sync response
+    /// that contains a 'm.beacon' event.
+    pub fn subscribe_to_live_location_shares(
+        &self,
+    ) -> (JoinHandle<()>, broadcast::Receiver<LiveLocationShare>) {
+        let (sender, receiver) = broadcast::channel(16);
+
+        let client = self.client.clone();
+        let room_id = self.room_id().to_owned();
+
+        let handle: JoinHandle<()> = tokio::spawn(async move {
+            let beacon_event_handler_handle = client.add_room_event_handler(&room_id, {
+                move |event: OriginalSyncBeaconEvent| async move {
+                    let live_location_share = LiveLocationShare {
+                        user_id: event.sender,
+                        last_location: LastLocation {
+                            location: event.content.location,
+                            ts: event.content.ts,
+                        },
+                    };
+
+                    // Send the live location update to all subscribers.
+                    let _ = sender.send(live_location_share);
+                }
+            });
+
+            let _ = beacon_event_handler_handle;
+        });
+
+        (handle, receiver)
+    }
+}
+
+/// Details of the last known location beacon.
+#[derive(Clone, Debug)]
+pub struct LastLocation {
+    /// The most recent location content of the user.
+    pub location: LocationContent,
+    /// The timestamp of when the location was updated
+    pub ts: MilliSecondsSinceUnixEpoch,
+}
+
+/// Details of a users live location share.
+#[derive(Clone, Debug)]
+pub struct LiveLocationShare {
+    /// The user's last known location.
+    pub last_location: LastLocation,
+    // /// Information about the associated beacon event (currently commented out).
+    // pub beacon_info: BeaconInfoEventContent,
+    /// The user ID of the person sharing their live location.
+    pub user_id: OwnedUserId,
```

Comment:
Can you move these types into their own Rust module please?

Reply:
Moved these into a `live_location` module! Thanks for the review!